src/std_impls.rs

```diff
@@ -0,0 +1,25 @@
+// Bitcoin Numeric Library
+// Written in 2018 by
+//   Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+use hex;
+use std::error;
+
+impl error::Error for hex::Error {
```

Comment:
Is this allowed? I thought an impl had to be in the same module as either the type or the trait? Is it same crate instead of same module?

Reply:
`hex::Error` is defined in this module and you can impl foreign traits for locally-defined types. So yes, this works and compile.